aiobotocore/parsers.py

```diff
@@ -31,6 +33,43 @@ def _create_event_stream(self, response, shape):
         name = response['context'].get('operation_name')
         return AioEventStream(response['body'], shape, parser, name)
 
+    # Only JSONParser needs this to be async. Override it here at the base
+    # class so the interface is consistent on all aio response classes.
+    async def parse(self, response, shape):
+        LOG.debug('Response headers: %s', response['headers'])
+        LOG.debug('Response body:\n%s', response['body'])
+        if response['status_code'] >= 301:
+            if self._is_generic_error_response(response):
+                parsed = self._do_generic_error_parse(response)
+            elif self._is_modeled_error_shape(shape):
+                parsed = self._do_modeled_error_parse(response, shape)
+                # We don't want to decorate the modeled fields with metadata
+                return parsed
+            else:
+                parsed = self._do_error_parse(response, shape)
+        else:
+            parsed = self._do_parse(response, shape)
+            if inspect.iscoroutine(parsed):
+                parsed = await parsed
```

Comment:
The `iscoroutinefunction` checks previously in `endpoint.py` are handled here now.

Reply:
I love that you switched from deprecated `asyncio.iscoroutinefunction` to `inspect.iscoroutine`.

Reply:
I'll try to understand the difference of `iscouroutine` and `isawaitable` now. The helpers use `isawaitable`.
If I understand the [documentation](https://docs.python.org/3/library/inspect.html#inspect.iscoroutine) correct, `isawaitable` support functions decorated with `types.coroutine` and aren't defined by `async def`. If I understand that right, `isawaitable` covers a larger set of functions. 

I am going to change it to `_helpers.resolve_awaitable`.